lib/Sema/ConstraintGraph.cpp

```diff
@@ -1558,6 +1558,123 @@ void ConstraintGraph::dump(llvm::raw_ostream &out) {
   print(CS.getTypeVariables(), out);
 }
 
+void ConstraintGraph::dumpActiveScopeChanges(llvm::raw_ostream &out,
+                                             unsigned indent) {
+  if (Changes.empty() && (ActiveScope->getStartIdx() == Changes.size()))
```

Comment:
Is this supposed to be `||` instead of `&&`? Do we even need a second condition since the loop is from startIdx to size anyway?

Reply:
We no longer need the second condition and have removed it.